File: API-Proyecto/Domain/Interfaces/Repositories/IUsuarioRepository.cs
using Domain.Entities;

namespace Domain.Interfaces.Repositories
{
    /// <summary>
    /// Interfaz que define las operaciones del repositorio de Usuario.
    /// </summary>
    public interface IUsuarioRepository
    {
        /// <summary>
        /// Obtiene la lista completa de usuarios.
        /// </summary>
        /// <returns>Lista de usuarios</returns>
        List<Usuario> GetListaUsuarios();

        /// <summary>
        /// Obtiene un usuario por su identificador.
        /// </summary>
        /// <param name="idUsuario">ID del usuario a buscar</param>
        /// <returns>Usuario encontrado o null si no existe</returns>
        Usuario? GetUsuarioPorId(int idUsuario);

        /// <summary>
        /// Obtiene un usuario por su nombre.
        /// </summary>
        /// <param name="nombre">Nombre del usuario a buscar</param>
        /// <returns>Usuario encontrado o null si no existe</returns>
        Usuario? GetUsuarioPorNombre(string nombre);
    }
}

File: ProyectoGrupoSinco/src/domain/interfaces/repositories/IUsuarioRepository.ts
import { clsUsuario } from '../../entities/clsUsuario';

export interface IUsuarioRepository {
  getListaUsuarios(): Promise<clsUsuario[]>;
  getUsuarioPorId(idUsuario: number): Promise<clsUsuario>;
  getUsuarioPorNombre(nombre: string): Promise<clsUsuario>;
  crearUsuario(usuarioNuevo: clsUsuario): Promise<number>;
  actualizarUsuario(idUsuario: number, usuario: clsUsuario): Promise<number>;
  eliminarUsuario(idUsuario: number): Promise<number>;
}
